src/components/tl/sharp/tl_sharp_context.c

```diff
@@ -260,61 +381,70 @@ UCC_CLASS_INIT_FUNC(ucc_tl_sharp_context_t,
         goto err;
     }
 
-   if (UCC_OK != ucc_context_progress_register(
-                      params->context,
-                      (ucc_context_progress_fn_t)sharp_coll_progress,
-                      self->sharp_context)) {
-        tl_error(self->super.super.lib, "failed to register progress function");
-        status = UCC_ERR_NO_MESSAGE;
-        goto err_clean_mpool;
-   }
-
-   if (self->cfg.use_rcache) {
-       rcache_params.alignment          = 64;
-       rcache_params.ucm_event_priority = 1000;
-       rcache_params.max_regions        = ULONG_MAX;
-       rcache_params.max_size           = SIZE_MAX;
-       rcache_params.region_struct_size = sizeof(ucc_tl_sharp_rcache_region_t);
-       rcache_params.max_alignment      = getpagesize();
-       rcache_params.ucm_events         = UCM_EVENT_VM_UNMAPPED |
-                                          UCM_EVENT_MEM_TYPE_FREE;
-       rcache_params.context            = self;
-       rcache_params.ops                = &ucc_tl_sharp_rcache_ops;
-       rcache_params.flags              = 0;
-
-       status = ucc_rcache_create(&rcache_params, "SHARP", &self->rcache);
-       if (status != UCC_OK) {
-           tl_error(self->super.super.lib, "failed to create rcache");
-           status = UCC_ERR_NO_RESOURCE;
-           goto err_clean_progress;
-       }
+    if (self->cfg.use_rcache) {
+        rcache_params.alignment          = 64;
+        rcache_params.ucm_event_priority = 1000;
+        rcache_params.max_regions        = ULONG_MAX;
+        rcache_params.max_size           = SIZE_MAX;
+        rcache_params.region_struct_size = sizeof(ucc_tl_sharp_rcache_region_t);
+        rcache_params.max_alignment      = getpagesize();
+        rcache_params.ucm_events         = UCM_EVENT_VM_UNMAPPED |
+                                            UCM_EVENT_MEM_TYPE_FREE;
+        rcache_params.context            = self;
+        rcache_params.ops                = &ucc_tl_sharp_rcache_ops;
+        rcache_params.flags              = 0;
+
+        status = ucc_rcache_create(&rcache_params, "SHARP", &self->rcache);
+        if (status != UCC_OK) {
+            tl_error(self->super.super.lib, "failed to create rcache");
+            status = UCC_ERR_NO_RESOURCE;
+            goto err_clean_mpool;
+        }
    }
 
    tl_info(self->super.super.lib, "initialized tl context: %p", self);
    return UCC_OK;
 
-err_clean_progress:
-    ucc_context_progress_deregister(
-        params->context, (ucc_context_progress_fn_t)sharp_coll_progress,
-        self->sharp_context);
 err_clean_mpool:
     ucc_mpool_cleanup(&self->req_mp, 0);
 err:
     return status;
 }
 
+ucc_status_t ucc_tl_sharp_context_create_epilog(ucc_base_context_t *context)
+{
```

Comment:
is epilog needed in order to internal OOB to be ready?

Reply:
yes, Internal OOB during ctx creation is based on service team. It will only be available during epilog